components/motor/gpio/motor_encoder.go

```diff
@@ -290,20 +267,15 @@ func sign(x float64) float64 { // A quick helper function
 // DirectionMoving returns the direction we are currently moving in, with 1 representing
 // forward and  -1 representing backwards.
 func (m *EncodedMotor) DirectionMoving() int64 {
-	m.stateMu.RLock()
-	defer m.stateMu.RUnlock()
+	m.mu.RLock()
+	defer m.mu.RUnlock()
 	return int64(m.directionMovingInLock())
 }
 
 func (m *EncodedMotor) directionMovingInLock() float64 {
 	move, err := m.real.IsMoving(context.Background())
 	if move {
-		_, isSingle := m.encoder.(*single.Encoder)
-		if sign(m.state.lastPowerPct) != sign(m.state.direction) && isSingle {
-			// short sleep when changing directions to minimize lost ticks in single encoder
-			time.Sleep(10 * time.Microsecond)
-		}
-		return sign(m.state.lastPowerPct)
```

Comment:
actually I think we can delete this. I just tried spamming everything and didn't see any issues, I'm assuming general clean up of both rpmMonitor and loop helped this in some way. So if we delete this I don't need to re-add direction to the struct

Reply:
@randhid whenever you have time could you try to spam bucket rover for me because you had the magic touch last time with getting it to freak out